R/module_nested_tabs.R

```diff
@@ -371,7 +364,7 @@ srv_teal_module.teal_module <- function(id,
         modules = modules
       )
 
-      summary_table <- srv_data_summary("data_summary", module_teal_data)
+      summary_table <- srv_data_summary("data_summary", summary_data) # Only updates on success
```

Comment:
I think this should not be in error state or hidden when transformators introduce an error.

The alternative is to use `module_teal_data` and remove the (newly) introduced `summary_data` as below.

Q: WDYT?

![image](https://github.com/user-attachments/assets/2d38b967-2a08-404c-bb2e-e57103ee5a98)